test/gateway_composer/utils.js

```diff
@@ -12,23 +12,118 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-const GatewayComposer = artifacts.require('TestGatewayComposer');
+const BN = require('bn.js');
 
-/**
- * Setup GatewayComposer.
- */
-module.exports.setupGatewayComposer = async (accountProvider) => {
+const MockBrandedTokenPass = artifacts.require('MockBrandedTokenPass');
+const MockBrandedTokenFail = artifacts.require('MockBrandedTokenFail');
+const GatewayComposer = artifacts.require('GatewayComposer');
+const EIP20TokenMock = artifacts.require('EIP20TokenMock');
```

Comment:
Might it make sense to use mock EIP20 contracts similar to what is done to mock Gateway and BT functionality?

I realize it is more files, but I think it would also be "easier" (i.e., not having to set balances or risking that incorrect values make testing more difficult).

Food for thought 🤔

Reply:
Thinking 🤔 more on it there are test cases where we needed to set actual balance and validate it.
e.g. Validate BrandedToken ValueToken balance in GC+BT requestStake

Reply:
I disagree, for the same reason as I do not think we should test what happens at the destination for the `transferToken` function: the value token is meant to be an external contract over which we have no control or information (however, if my understanding of value token is incorrect, then this view may not be sensible).

Because we cannot trust the value token contract, we might as well save ourselves the hassle / lines of code that come from using a realistic token contract in testing and have only what's necessary to get returns in order to test the responses of our contracts.